### DISCUSSION TITLE
Backend SSO

### DISCUSSION THREAD

**Discussion by JMillwood-92** [pretix/pretix#4292] [Ideas]
Hello, 
I apologise if this has already been posed as a suggestion or question, but I couldn't find a thread specific to this idea / question when I searched to forums or wider via google.

I currently have a active directory of users, with SSO used across all of our online services, except for Pretix. 

I'm aware of the front-end SSO integration available, but is there any plans to add Single Sign On support for logging into the backend of Pretix?

**Comment by pc-coholic** [comment]
There are a few plugins available, have a look here: https://marketplace.pretix.eu/categories/9/

**Reply by JMillwood-92** [reply]
Thanks for your reply and suggestion. 
I had looked at these options, but none of them allow definition of a standard OAUTH provider for SSO, only the specific SSO service providers they were written for.
 
Pretix already has the functionality to add custom OAUTH/SSO providers in the customer accounts section, when you enable the Customer Accounts setting, you then get an SSO providers section, however this only works for mapping to customer authentication. 

I am essentially looking for identical functionality but that allows login to the Pretix backend. Primarily my events team user logs into their Microsoft Entra account, clicks a link to the Pretix / control webpage, and is automatically authenticated, because they are already logged into their tenant account. Pretix is provided a list of the users security groups, and matches the presence of security group to membership of an events team. 
Similar for example to the mini orange plugin for Wordpress, that allows authentication with a Wordpress site using Microsoft as an SSO provider.

**Reply by pc-coholic** [reply]
Building such a AuthenticationBackend should be possible - I am just not sure if anyone has done it yet...

https://github.com/adevolutio/pretix-oidc could probably work for generalized OAUTH-login from the pretix backend login page - but I haven't tested it myself.

While it should work in theory; I do not know if that specific implementation would allow to access directly the `redirect_url` of the pretix system and pass an authentication token...

But generally speaking: We did specifically build pretix to provide a space for AuthenticationBackends to plug into, since we do not want to ship a bunch of them ourselves. Doing proper authentication is a complicated enough as is, and more often than not a "one for all"-plugin will not exist or will be quite extensive and difficult to properly configure. So I don't think we'll see a pretix maintained and shipped OAUTH-Backend-Auth provider shipped any time soon, but we'll be happy to see the above linked one being fleshed out :-)
